### PR TITLE
[13.0][IMP+FIX] account_asset_management: Add chatter entry in invoice for generated asset, FIX reference empty

### PR DESCRIPTION
**Pull request by ValentinVinagre**
**IMP**:
Coming from: https://github.com/OCA/account-financial-tools/issues/1006
Added chatter entry with the link to the asset.

**FIX** (empty asset reference):
If the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

### CODE REVIEW

**Comment by ValentinVinagre** [comment]
@HaraldPanten could you review it?
Thanks

File: account_asset_management/models/account_move.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ def write(self, vals):
         return super().write(vals)
 
     def post(self):
+        super().post()
```
Comment (pedrobaeza) [inline account_asset_management/models/account_move.py:62]: Why this change?

Comment (ValentinVinagre) [reply account_asset_management/models/account_move.py:62]: Because if the assets are created before posting the invoice, they wouldn't have a name and the "code" field of the assets will be empty, but they would have the invoice name.

**Review by HaraldPanten — APPROVED** [review]
It LGTM 👍

File: account_asset_management/models/account_move.py (hunk 1 above)
Comment (pedrobaeza) [reply account_asset_management/models/account_move.py:62]: But that is another thing. Please always split these things is separate PRs for having separate debate. I'm also suspecting that the culprit of that is our module `l10n_es_account_invoice_sequence`, not this module.

Comment (HaraldPanten) [reply account_asset_management/models/account_move.py:62]: > But that is another thing. Please always split these things is separate PRs for having separate debate. I'm also suspecting that the culprit of that is our module `l10n_es_account_invoice_sequence`, not this module.

I don't think so. As we are working for V13 and l10n_es_account_invoice_sequence does not exist for V13. Might you be talking about V12?

Comment (pedrobaeza) [reply account_asset_management/models/account_move.py:62]: Yes, I was talking in v12, that the operation is performed that way. Then it can be of this module, but please put a separate PR with this fix (that doesn't affect the chatter entry), and explain in detail the problem in bot the PR and the commit (`FIX reference empty`) is not enough explanation.

Comment (HaraldPanten) [reply account_asset_management/models/account_move.py:62]: Just to know... Would it be OK to create a single PR but several commits for different FIXs and IMPs as long as each was well explained?

Comment (pedrobaeza) [reply account_asset_management/models/account_move.py:62]: That can serve, but take into account that if one of the changes generate debate, it will delay the merge of the other features that can be very clear, so you have to balance this for deciding if putting one PR or several.

Comment (ValentinVinagre) [reply account_asset_management/models/account_move.py:62]: @pedrobaeza as the IMP depends on the FIX, can I do 2 commits?

Comment (pedrobaeza) [reply account_asset_management/models/account_move.py:62]: I don't think they depend each other, but you get a "better" chatter entry when the FIX will arrive, but OK.

Comment (ValentinVinagre) [reply account_asset_management/models/account_move.py:62]: Done

**Comment by pedrobaeza** [comment]
You are still explaining nothing in the commit message, and the super was in the middle by a reason. @luc-demeyer can you tell us why it was that way?

**Comment by ValentinVinagre** [comment]
> You are still explaining nothing in the commit message, and the super was in the middle by a reason. @luc-demeyer can you tell us why it was that way?

I changed the last commit and the header of the PR.

**Review by HaraldPanten — APPROVED** [review]
LGTM 👍

File: account_asset_management/models/account_move.py [hunk 2]
```diff
@@ -79,7 +80,15 @@ def post(self):
...
                 aml.with_context(allow_asset=True).asset_id = asset.id
-        super().post()
+            refs = [
+                "<a href=# data-oe-model=account.asset data-oe-id=%s>%s</a>"
+                % tuple(name_get)
+                for name_get in move.line_ids.filtered(
+                    "asset_profile_id"
+                ).asset_id.name_get()
+            ]
+            message = _("This invoice created the asset/s: %s") % ", ".join(refs)
```
Comment (luc-demeyer) [inline account_asset_management/models/account_move.py]: I think we should change 'asset/s' into 'asset(s)'

Comment (ValentinVinagre) [reply account_asset_management/models/account_move.py]: Done!

**Review by luc-demeyer — APPROVED** [review]
